test/helpers/domain/domain.ts

```diff
@@ -0,0 +1,287 @@
+        tokenOwner,
+        tokenURI,
+        distrConfig,
+        paymentConfig,
+      });
+    } else {
+      txPromise = await zns.subRegistrar.connect(executor ? executor : owner).registerSubdomain({
+        parentHash,
+        label,
+        domainAddress: hre.ethers.isAddress(domainAddress) ? domainAddress : owner.address,
+        tokenOwner,
+        tokenURI,
+        distrConfig,
+        paymentConfig,
+      });
+    }
+
+    this.hash = await this.getDomainHashFromEvent(executor);
+
+    return txPromise;
+  }
+
+  async revoke (executor ?: SignerWithAddress) : Promise<ContractTransactionResponse> {
+    return this.zns.rootRegistrar.connect(executor ? executor : this.owner).revokeDomain(this.hash);
+  }
+
+  async assignDomainToken (
+    to : string,
+    executor ?: SignerWithAddress
+  ) : Promise<ContractTransactionResponse> {
+    return this.zns.rootRegistrar.connect(executor ? executor : this.owner).assignDomainToken(
+      this.hash,
+      to
+    );
+  }
+
+  async updateDomainRecord (
+    resolverType : string,
+    executor ?: SignerWithAddress
+  ) : Promise<ContractTransactionResponse> {
+    return this.zns.registry.connect(executor ? executor : this.owner).updateDomainRecord(
+      this.hash,
+      this.owner,
+      resolverType,
+    );
+  }
+
+  async updateMintlistForDomain (
+    candidates : Array<string>,
+    allowed : Array<boolean>,
+    executor ?: SignerWithAddress
+  ) : Promise<ContractTransactionResponse> {
+    if (candidates.length !== allowed.length)
+      throw new Error("Domain Helper: Candidates and allowed arrays must have the same length");
+
+    return this.zns.subRegistrar.connect(executor ? executor : this.owner).updateMintlistForDomain(
+      this.hash,
+      candidates,
+      allowed,
+    );
+  }
+
+  // ------------------------------------------------------
+  // SETTERS
+  // ------------------------------------------------------
+  async setOwnersOperator (
+    operator : string,
+    allowed : boolean,
+    executor ?: SignerWithAddress
+  ) : Promise<ContractTransactionResponse> {
+    return this.zns.registry.connect(executor ? executor : this.owner).setOwnersOperator(operator, allowed);
+  }
+
+  async setDistributionConfigForDomain (
+    executor ?: SignerWithAddress
+  ) : Promise<ContractTransactionResponse> {
+    return this.zns.subRegistrar.connect(executor ? executor : this.owner).setDistributionConfigForDomain(
+      this.hash,
+      this.distrConfig,
+    );
+  }
+
+  async setPricerDataForDomain (
+    priceConfig ?: ICurvePriceConfig | IFixedPriceConfig,
+    pricerContract ?: string,
+    executor ?: SignerWithAddress
+  ) : Promise<ContractTransactionResponse> {
+    return this.zns.subRegistrar.connect(executor ? executor : this.owner).setPricerDataForDomain(
+      this.hash,
+      priceConfig ? encodePriceConfig(priceConfig) : encodePriceConfig(this.priceConfig),
```

Comment:
This would throw an error if the incoming `priceConfig` param is `undefined` and the default case above is hit in your switch statement where `priceConfig === {}`

Resolve the `priceConfig` to use before this return statement, and if the incoming param is `undefined` and `this.priceConfig` is `{}`